src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,195 @@
+
+@Log4j2
+public class FixedTokenLengthChunker implements IFieldChunker {
+
+    // parameters
+
+    public static final String TOKEN_LIMIT = "token_limit";
+    public static final String OVERLAP_RATE = "overlap_rate";
+    public static final String MAX_TOKEN_COUNT = "max_token_count";
+    public static final String TOKENIZER = "tokenizer";
+
+    // default values for each parameter
+    private static final int DEFAULT_TOKEN_LIMIT = 500;
+    private static final double DEFAULT_OVERLAP_RATE = 0.2;
+    private static final int DEFAULT_MAX_TOKEN_COUNT = 10000;
+    private static final String DEFAULT_TOKENIZER = "standard";
+
+    private final AnalysisRegistry analysisRegistry;
+
+    public FixedTokenLengthChunker(AnalysisRegistry analysisRegistry) {
+        this.analysisRegistry = analysisRegistry;
+    }
+
+    private List<String> tokenize(String content, String tokenizer, int maxTokenCount) {
+        AnalyzeAction.Request analyzeRequest = new AnalyzeAction.Request();
+        analyzeRequest.text(content);
+        analyzeRequest.tokenizer(tokenizer);
+        try {
+            AnalyzeAction.Response analyzeResponse = analyze(analyzeRequest, analysisRegistry, null, maxTokenCount);
+            List<AnalyzeAction.AnalyzeToken> analyzeTokenList = analyzeResponse.getTokens();
+            List<String> tokenList = new ArrayList<>();
+            for (AnalyzeAction.AnalyzeToken analyzeToken : analyzeTokenList) {
+                tokenList.add(analyzeToken.getTerm());
+            }
+            return tokenList;
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static int getOverlapTokenNumber(int tokenCount, double overlapRate) {
+        return Math.min(tokenCount - 1, (int) Math.floor(tokenCount * overlapRate));
+    }
+
+    private static int getDocumentTokenCount(int tokensPerPassage, int passageCount, double overlapRate) {
+        // return the token count of the document, if all (passageCount) passages have (tokensPerPassage) tokens
+        int overlapTokenNumber = getOverlapTokenNumber(tokensPerPassage, overlapRate);
+        return tokensPerPassage + (passageCount - 1) * (tokensPerPassage - overlapTokenNumber);
+    }
+
+    private static int getPassageCount(int contentLength, int tokenLimit, double overlapRate) {
+        /*
+        passageCount means the number of chunked passages, which should be the minimum integer
+        so that getDocumentTokenCount(tokenLimit, passageCount, overlapRate) >= contentLength
+        */
+        int overlapTokenNumber = getOverlapTokenNumber(tokenLimit, overlapRate);
+        return 1 + (int) Math.ceil((contentLength - tokenLimit) / (double) (tokenLimit - overlapTokenNumber));
+    }
+
+    private static int getTokensPerPassage(int contentLength, int passageCount, int tokeLimit, double overlapRate) {
+        /*
+        To evenly chunk the documents, the token length difference among passages is at most 1.
```

Comment:
Is this evenly chunk a hard requirement? What's the benefit of doing this?

Reply:
Evenly chunk will not produce a short passage at the last.

Reply:
By evenly chunk, the last chunked passage will not be a very short passage.

Reply:
I have reverted the code to undo the evenly chunk feature.